geps/gep-1619.md

```diff
@@ -130,98 +275,604 @@ Generally, the implementation API programs the dataplane API; however these two
+considered cautiously, as making associations to Gateway API's notion of Gateway, Route, and Service to other
+implementation's objects is hard to directly translate. The idea of a route in Gateway API is often not the same as a
+route in any given implementation.
+
+### Metaresource Policy Design
+
+In order to apply session persistence configuration to both a service and a route, we will implement it as a
+metaresource policy, as outlined in [Policy Attachment](https://gateway-api.sigs.k8s.io/references/policy-attachment/#direct-policy-attachment).
+The metaresource is named `SessionPersistencePolicy` and is only responsible for configuring session persistence for
+services or routes. It is defined as a [Direct Policy Attachment](https://gateway-api.sigs.k8s.io/v1alpha2/references/policy-attachment/#direct-policy-attachment)
+without defaults or overrides, applied to the targeted service, HTTPRoute, or GRPCRoute.
+
+Attaching the `SessionPersistencePolicy` metaresource to a service will be a core support level feature, while attaching
+it to a route will be considered extended. Implementations must support services for conformance, while routes are
+considered optional. This distinction arises because most existing implementations primarily support attaching to a
+service, while attaching to a route is less common and involves greater complexities (see [API Attachment Points](#api-attachment-points)
+for more details).
+
+### API Attachment Points
+
+The `SessionPersistencePolicy` metaresource can target both a service and route object. The core functionality of
+attaching it to either is generally identical. The distinction lies in where the configuration gets propagated: when
+attached to a route, it can define session persistence for multiple services, whereas attaching it to a service defines
+it for a single service. Enabling session persistence doesn't require configuration on both the route and service;
+configuring it on either one is sufficient.
+
+Given that certain implementations will only support attaching to services, and considering that some implementations
+might want to solely support attachment to routes, implementations must provide support for at least one of the options
+(route or service) to be compliant with this GEP.
+
+Edge cases will arise when providing supporting both routes and services. For guidance on addressing conflicting
+attachments, please consult the [Expected API Behavior](#expected-api-behavior) section, which outlines API use cases.
+
+We are anticipating the possibility of future enhancements in supporting the selection of the route rule (e.g. [HTTPRouteRule](https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io%2fv1beta1.HTTPRouteRule))
+section within xRoutes, contingent upon updates to [GEP-713](/geps/gep-713) for incorporating section targeting (PR [#2283](https://github.com/kubernetes-sigs/gateway-api/pull/2283)).
```

Comment:
Note that PR [#2283](https://github.com/kubernetes-sigs/gateway-api/pull/2283) is merged so the text saying "contingent" should be updated. I think we should say something like this:

supporting the selection of the route rule (e.g. [HTTPRouteRule](https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io%2fv1beta1.HTTPRouteRule)) section within xRoutes, is specified in  [GEP-713](https://github.com/kubernetes-sigs/gateway-api/blob/main/geps/gep-713.md#apply-policies-to-sections-of-a-resource) for incorporating section targeting.

(In fact I am wondering whether we should update this doc to expand the attachment points to include section targeting. But I understand this can cause unnecessary delays and we would like to expedite approving and merging this PR, so I am okay with leaving this here. However the fact that PR #2283 is now merged should be reflected here).

Reply:
Good catch. I've cleaned up this section to add support for all three - hopefully that won't be too contentious as I think we all agreed on this direction. Let me know what you think.